### ISSUE TITLE
Case insensitive keywords

### ISSUE THREAD

**Issue by subeax** [nepsilon/search-query-parser#22]
Hi!
Thank you for this amazing library!

I think it may be helpful to ignore keywords case.

    const options = {
	    keywords: ['name', 'town'],
    };

    const query = "Name:John,Ben,Bill";

might be done by adding new option, like 

    ignoreCase: true // or something else

and adding condition to line 130:

    // We got an advanced search syntax
    else {
     key = options.ignoreCase 
              ? term.keyword.toLowerCase() 
              :  term.keyword
    ...

Thoughts? :)

**Comment by nepsilon** [comment]
Yes, very good idea! Can you send a pull request with this addition and a few tests?
